### ISSUE TITLE
Feature Request: Improved exclusion of files

### ISSUE THREAD

**Issue by slusarz** [jooy2/vitepress-sidebar#145]
Use-case: we have a centralized "core" documentation repository that is used by several projects.

Certain pages should be hidden for each of those projects.

Vitepress's _srcExclude_ config item allows us to provide a list of the pages to be excluded, and vitepress will not process/bundle those pages.  (This list is programmatically created by searching for a specific, locally-defined frontmatter parameter.)

We use vitepress-sidebar to create the sidebar, so we also need to remove the page from the sidebar list.  We can't use 'exclude' frontmatter, since this page is not always hidden (depends on the project being built)

Problem: It would be great if we could pass the same list of files from _srcExclude_ to vitepress-sidebar's _excludeFiles_ option.  Unfortunately, we discovered that **_excludeFiles_ ONLY matches the filename, not the full path**.  So this doesn't work if there are duplicate filenames living in the tree.

Example: I want to hide `/path/to/index.md`, but I have 57 other index.md files living in the tree.  There is no way to hide `/path/to/index.md` currently with _excludeFiles_ without hiding the other pages, since it only matches on the entry `index.md`.

Solution: Match on full pathnames.  (More elaborate solution: merge _excludeFiles_ and _excludeFolders_ and have a single exclusion config item, like .vitepress _srcExclude_, that uses fast-glob - since that captures both the individual file and the folder exclusion use-cases.)

Alternative Solution: Add a callback function (e.g. _excludeCallback_) that passes the page name and frontmatter to the callback, and allow the callback to return true/false whether to exclude the page.

Solution: Apologies there is no MR here, but since this could be implemented in a number of different ways it would be premature to decide a solution at this time.

**Comment by Peque** [comment]
Yeah, it would be great if `vitepress-sidebar` could honor `srcExclude`. That way, the exclusion needs to be defined only once. :blush:

**Comment by javinba** [comment]
I agree, it would be great!

**Comment by jooy2** [comment]
Hello guys! Thank you for using `vitepress-sidebar`.

Thank you for your patience. I have added a new option `excludePattern` that allows you to exclude files and folders using glob patterns. In the next major version, we plan to remove the `excludeFiles` and `excludeFolders` options accordingly.

Also you need to upgrade `vitepress-sidebar` to `1.26.0`.

This is not a heavily tested feature yet, so you may experience issues with the behavior. If this is the case, please create a new issue and we'll mark it as closed since it covers a feature request.

Regards,

**Comment by Peque** [comment]
@jooy2 Thanks! It seems to work just fine for my use case. :+1: :blush:

**Comment by Peque** [comment]
PS: I created https://github.com/jooy2/vitepress-sidebar/issues/181 as a follow-up to allow inheriting/honoring `srcExclude` from VitePress config (i.e.: avoid the need to declare the same exclusion in `excludePattern` at the plugin level if it has already been declared in `srcExclude` at the VitePress level).
